ethcore/evm/src/interpreter/mod.rs

```diff
@@ -227,13 +233,7 @@ impl<Cost: CostType> Interpreter<Cost> {
 			((instruction == instructions::SHL || instruction == instructions::SHR || instruction == instructions::SAR) && !schedule.have_bitwise_shifting) {
 
 			return Err(vm::Error::BadInstruction {
-				instruction: instruction
-			});
-		}
-
-		if info.tier == instructions::GasPriceTier::Invalid {
-			return Err(vm::Error::BadInstruction {
-				instruction: instruction
+				instruction: instruction as u8
```

Comment:
nit: Could use `opcode` instead of `instruction as u8`.

Reply:
We don't have opcode variable in `verify_instruction`. Because `instruction as u8` is basically a noop, I think this would be okay?